### ISSUE TITLE
INSTINCT for white box LLM

### ISSUE THREAD

**Issue by Monoclinic** [xqlin98/INSTINCT#1]
Hello, thanks for your nice work and I really learned a lot from your paper.
I have a question, INSTINCT is designed for black box LLM like chatGPT. 
If I have an open-source LLM of size 6~7B, and I have some data to tune it myself, I wonder if the performance of INSTINCT would be the same as (or surpass) those methods based on gradients, such as p-tuning as mentioned in your paper (page 2 line 2).

**Comment by xqlin98** [comment]
Thank you for your interest in our paper. Our INSTINCT algorithm is designed for the black-box model setting, where access to the model's weights is not possible. Our goal is to identify the best instruction with as few queries to the black-box model as possible, aiming for query efficiency. In scenarios where the LLM is a white-box model, thus not constrained by a query budget, our INSTINCT algorithm (a global optimization algorithm) theoretically could outperform gradient-based approaches (local optimization algorithms) given an unlimited query budget. However, empirically, the query budget is never unlimited, as we typically run a fixed number of iterations. Therefore, we cannot conclusively determine whether our approach performs better than gradient-based approaches empirically, as we have not conducted experiments in those settings. Please let me know if you have any further questions.

**Comment by Monoclinic** [comment]
Thanks for your kind reply.
Can I say INSTINCT is theoretically better than gradient-based methods because INSTINCT can be tuned directly on validation set, and if given sufficient steps for exploration it can always achieve a best prompt. However those gradient-based methods are trained on training set, yet there is always a domain gap between training and test set, so the prompt would not be a perfect one for validation.

Hmm...I reflect on my words above. Maybe it's not so reasonable... =_=||
So why INSTINCT is theoretically better?

**Comment by xqlin98** [comment]
INSTINCT uses the neural bandits algorithm, which is a global optimization algorithm. In other words, the neural bandits algorithm provides a theoretical guarantee that as the number of iterations **approaches infinity**, the best function value found will converge to the global optimum. In contrast, gradient descent approaches only guarantee convergence to a local optimum. In this sense, the neural bandits algorithm is superior to gradient-based approaches theoretically.

**Comment by Monoclinic** [comment]
I see. Thank you for your clear and helpful explanation.
